imageio/plugins/pillow.py

```diff
@@ -101,10 +104,17 @@ def _can_read(self, request):
         Image = self._init_pillow()
         if request.mode[1] in (self.modes + "?"):
             if self.plugin_id in Image.OPEN:
-                factory, accept = Image.OPEN[self.plugin_id]
+                try:
+                    factory, accept = Image.OPEN[self.plugin_id]
+                except Exception:  # can hapen, see e.g. #588
+                    factory, accept = None, None
                 if accept:
                     if request.firstbytes and accept(request.firstbytes):
                         return True
+                elif factory:
+                    ext = request.filename.split(".")[-1].lower()
+                    if ext in FORMATS_ACCEPTED_BY_DEFAULT:
+                        return True
```

Comment:
This is the part of interest. The rest is a result of running the autoformatter.